pkg/provider/vault/symlink_test.go

```diff
@@ -0,0 +1,66 @@
+package vault
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"testing"
+)
+
+func TestIsSymlink(t *testing.T) {
+	cases := map[string]struct {
+		got  string
+		want bool
+	}{
+		"ShouldResolveSymlink": {
+			got:  "vault://test",
+			want: true,
+		},
+		"ShouldNotResolveSymlink": {
+			got:  "test",
+			want: false,
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			if diff := cmp.Diff(tc.want, isSymlink(tc.got), EquateErrors()); diff != "" {
+				t.Errorf(diff)
+			}
+		})
+	}
+}
+
+func TestExtractSymlinkParts(t *testing.T) {
```

Comment:
If there are invalid cases we should add these case here as well.

Reply:
This function will not be reached if the secret don't match the above regex. I will enrich the test cases with more examples